### ISSUE TITLE
Localsend on iPadOS does not run in background

### ISSUE THREAD

**Issue by freebrowser1** [localsend/localsend#1468]
### Describe the bug

On an iPad Air I installed this awesome app and I was able to send and receive files from other devices: Macbook Pro and Android phone using the same localsend.
I enabled: "Quick save", "Auto Finish" and "Share via link: Auto accept" on the iPad.

But when I send a file to the iPad and the app is in the background, the sender has to wait on the iPad which is not the case when sending to Android or macOS. 
When I refresh the list of 'Available devices' on the other device, the iPad is removed from the list, so it is not found when in the background, even when the screen is on.
Is that a limitation of iOS which appears to not supporting multitasking / running apps in the background ?

### To reproduce

1. Start localsend on iPad and open another app on it so that localsend moves to the background.
2. Send a file from localsend from another device (non-iOS, e.g.macOS or Android).
3. The sending device has to wait until the iPad app is on the foreground.

### Expected behavior

It should run in the background (does iOS allow this ?).

### Desktop (please complete the following information)

- OS: iPadOS 17.x
- Device : iPad Air

**Comment by mak448a** [comment]
I think this is a limitation of iPadOS.

**Comment by freebrowser1** [comment]
You are right, but some apps, like iOS mail apps or IM apps like Elements do have the possibility that a message is ready.

**Comment by mak448a** [comment]
You are correct. But I’m pretty sure that’s server based, which this app is not.

**Comment by mak448a** [comment]
Supposedly, you can keep an app running with PiP or location access. Just an idea.

**Comment by freebrowser1** [comment]
> Supposedly, you can keep an app running with PiP or location access. Just an idea.
Indeed, I know of an app (which I tried) called iSH which is allowed in the App Store and can be run in the background, even as an SSH server. This app uses location access. I don't know what PiP means.
That makes this app useful under iOS as under other platforms it does run in the background which is more or less a requirment for a file transfer app.

**Comment by mak448a** [comment]
> > Supposedly, you can keep an app running with PiP or location access. Just an idea.
> 
> Indeed, I know of an app (which I tried) called iSH which is allowed in the App Store and can be run in the background, even as an SSH server. This app uses location access. I don't know what PiP means. That makes this app useful under iOS as under other platforms it does run in the background which is more or less a requirment for a file transfer app.

PiP is picture in picture (that feature of YouTube that allows you to keep watching a video outside the app). You can play a video continuously to achieve background playback.

**Comment by Clarkkkk** [comment]
With [iOS 26](https://developer.apple.com/videos/play/wwdc2025/227/), I think it is possible now.

**Comment by EliasGabr** [comment]
i have noticed that on ios many apps use the dynamic island (live activities) to finish stuff in the background. Maybe that could be used to enable background activities on ios

**Comment by Null-698** [comment]
With iOS 26 (live activities), the app can run in the background and on the lock screen, and it’s super easy to set up. I’ve added it to my private fork, and it took about 15 minutes with AI. Of course, I won’t be using the AI generated code/slop to contribute. So, please add this feature so I can use the app without worrying about the annoying 7day refresh.

**Comment by Tienisto** [comment]
@Null-698 can you share me the fork? I will check if the implementation seems reasonable.

**Comment by Null-698** [comment]
@Tienisto Here you go https://github.com/Null-698/localsend

I had to dig deep through my commits to find it, lol, so I reforked the repository and pushed the changes. This should make everything easier to review. I also added some comments and tried to reorganize the code as much as I could.

also you can safely remove the sideStore suffix workaround. For some reason, wildcard identifiers do not work with BG tasks when the app is sideloaded, so I had to add that workaround.

**EDIT:** patching sidestore to rewrite the background task identifiers from the original bundle id to the final signed bundle id before signing fixed the issue. this goes beyond localsend, so sidestore, iloader, and similar signing tools should patch it on their end.
